### PR TITLE
Add decorator to handle exceptions for commands

### PR DESCRIPTION
**Pull request by gmazoyer**
Fixes https://github.com/opsmill/infrahub/issues/3506

Commands to use decorator with:

- [x] `infrahubctl branch`
- [x] `infrahubctl check`
- [x] `infrahubctl generator`
- [x] `infrahubctl render`
- [x] `infrahubctl run`
- [x] `infrahubctl schema`
- [x] `infrahubctl transform`
- [x] `infrahubctl version`

The decorator code is redundant due to having to handle both sync and async commands. Is it worth to have async commands since these have a pretty linear execution?

### CODE REVIEW

**Review by ajtmccarty — APPROVED** [review]

File: python_sdk/infrahub_sdk/ctl/utils.py [hunk 1]
```diff
@@ -24,6 +35,69 @@ def init_logging(debug: bool = False) -> None:
     logging.getLogger("infrahubctl")
 
 
+def catch_exception(
+    which_exception: Union[type[Exception], Iterable[type[Exception]]] = Exception,
+    console: Console = None,
```
Comment (ogenstad) [inline python_sdk/infrahub_sdk/ctl/utils.py]: This feels like something that mypy would complain about, might be that the rules are slightly incorrect.

**Review by ogenstad — APPROVED** [review]
LGTM. We've talked about providing different output options for infrahubctl in the future. I.e. json output etc. With that in mind it could be that we need to revisit this part later. I think this is fine for now though. What I've been thinking is that we'd pass around some object with the same interface/protocol and by default our object would use `console` under the hood but it would also be possible to switch it out to use JSON or other formats. When we get there we might need to change how these decorators work. Mostly my thoughts about the future and nothing that needs to stop this PR.

**Comment by ogenstad** [comment]
Also if you rebase this PR it should fix the flaky end to end tests.

File: python_sdk/infrahub_sdk/ctl/utils.py (hunk 1 above)
Comment (gmazoyer) [reply python_sdk/infrahub_sdk/ctl/utils.py]: image

Comment (ogenstad) [reply python_sdk/infrahub_sdk/ctl/utils.py]: :)

**Pull request merged**